Examples/Modules/laser_injection/analysis_2d.py

```diff
@@ -0,0 +1,199 @@
+#! /usr/bin/env python
+
+# Copyright 2019 Andrew Myers, Jean-Luc Vay, Maxence Thevenet
+# Remi Lehe, Weiqun Zhang, Luca Fedeli
+#
+# This file is part of WarpX.
+#
+# License: BSD-3-Clause-LBNL
+
+# This file is part of the WarpX automated test suite. Its purpose is to test the
+# injection of a Gaussian laser pulse from an antenna in a 2D simulation.
+# In order to avoid privileged directions, the laser is injected at
+# approximately 27 degrees with respect to the x axis. Moreover the polarization axis is neither
+# parallel nor perpendicular to the xz plane. Finally moving window along the
+# x axis is enabled.
+# The test calculates the envelope of each component of the laser pulse at the end of
+# the simulation and it compares it with theory. It also checks that the
+# central frequency of the Fourier transform is the expected one.
+
+import yt
+import sys
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.signal import hilbert
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+import checksumAPI
+
+# Maximum acceptable error for this test
+relative_error_threshold = 0.05
+
+# A small number
+small_num = 1.0e-8
+
+# Physical parameters
+um = 1.e-6
+fs = 1.e-15
+c = 299792458
+
+# Parameters of the gaussian beam
+wavelength = 1.*um
+w0 = 5.*um
+tt = 10.*fs
+x_c = 10.*um
+t_c = 20.*fs
+# foc_dist = 13.109*um (not actually used)
+E_max = 4e12
+
+# laser direction
+dir_vector = np.array([2.,0,1.0])
+dir_vector /= np.linalg.norm(dir_vector)
+
+rot_angle = np.arctan(dir_vector[2]/dir_vector[0])
+
+# polarization vector
+pol_vector = np.array([1.0,1.0,-2.0])
+pol_vector /= np.linalg.norm(pol_vector)
+
+# Simulation parameter
+steps=220
+
+# Calculates the envelope of a Gaussian beam
+def gauss_env(T,XX,ZZ):
+    '''Function to compute the theory for the envelope
+    '''
+
+    X =  np.cos(rot_angle)*(XX-x_c) + np.sin(rot_angle)*ZZ
+    Z = -np.sin(rot_angle)*(XX-x_c) + np.cos(rot_angle)*ZZ
+
+    inv_tau2 = 1./tt/tt
+    inv_w_2 = 1.0/(w0*w0)
+    exp_arg = - (Z*Z)*inv_w_2 - inv_tau2 / c/c * (X-T*c)*(X-T*c)
```

Comment:
```suggestion
    Z =  np.cos(rot_angle)*(XX-x_c) + np.sin(rot_angle)*ZZ
    X = -np.sin(rot_angle)*(XX-x_c) + np.cos(rot_angle)*ZZ

    inv_tau2 = 1./tt/tt
    inv_w_2 = 1.0/(w0*w0)
    exp_arg = - (X*X)*inv_w_2 - inv_tau2 / c/c * (Z-T*c)*(Z-T*c)
```

Feel free to accept this change if you prefer to use the usual convention that Z is the propagation direction here.